--- a/gpMgmt/bin/gpconfig
+++ b/gpMgmt/bin/gpconfig
@@ -15,6 +15,7 @@
 import os
 import sys
 import re
+import shlex
 import psutil
 
 try:
@@ -382,7 +383,7 @@
 
     # Replace literal empty strings with empty quotes, or it will look like the
     # user passed in an incorrect argument, which would be misleading
-    params = [pipes.quote(arg) for arg in sys.argv[1:]]
+    params = [shlex.quote(arg) for arg in sys.argv[1:]]
     params = " ".join(params)
     if failure:
         LOGGER.error("finished with errors, parameter string '%s'" % params)
--- a/gpMgmt/bin/analyzedb
+++ b/gpMgmt/bin/analyzedb
@@ -22,7 +22,7 @@
 import time
 from datetime import datetime, timedelta
 from contextlib import closing
-import pipes  # for shell-quoting, pipes.quote()
+import shlex  # for shell-quoting, shlex.quote()
 import fcntl
 import itertools
 
@@ -950,7 +950,7 @@
 
 # Create a Command object that executes a query using psql.
 def create_psql_command(dbname, query):
-    psql_cmd = """psql %s -c %s""" % (pipes.quote(dbname), pipes.quote(query))
+    psql_cmd = """psql %s -c %s""" % (shlex.quote(dbname), shlex.quote(query))
     return Command(query, psql_cmd)
 
 
--- a/gpMgmt/bin/gppylib/operations/buildMirrorSegments.py
+++ b/gpMgmt/bin/gppylib/operations/buildMirrorSegments.py
@@ -1,6 +1,6 @@
 from contextlib import closing
 import os
-import pipes
+import shlex
 import signal
 import time
 import re
@@ -441,13 +441,13 @@
         if self.__progressMode != GpMirrorListToBuild.Progress.NONE:
             return GpMirrorListToBuild.ProgressCommand("tail the last line of the file",
                                                        "set -o pipefail; touch -a {0}; tail -1 {0} | tr '\\r' '\\n' |"
-                                                       " tail -1".format(pipes.quote(progressFile)),
+                                                       " tail -1".format(shlex.quote(progressFile)),
                                                        targetSegmentDbId, progressFile, ctxt=base.REMOTE,
                                                        remoteHost=targetHostname)
         return None
 
     def _get_remove_cmd(self, remove_file, target_host):
-        return base.Command("remove file", "rm -f {}".format(pipes.quote(remove_file)), ctxt=base.REMOTE, remoteHost=target_host)
+        return base.Command("remove file", "rm -f {}".format(shlex.quote(remove_file)), ctxt=base.REMOTE, remoteHost=target_host)
 
     def __runWaitAndCheckWorkerPoolForErrorsAndClear(self, cmds, suppressErrorCheck=False, progressCmds=[]):
         for cmd in cmds:
--- a/gpMgmt/bin/gppylib/commands/pg.py
+++ b/gpMgmt/bin/gppylib/commands/pg.py
@@ -4,7 +4,7 @@
 #
 
 import os
-import pipes
+import shlex
 
 from gppylib.gplog import *
 from gppylib.gparray import *
@@ -200,7 +200,7 @@
         # pg_rewind prints progress updates to stdout, but it also prints
         # errors relating to relevant failures(like it will not rewind due to
         # a corrupted pg_control file) to stderr.
-        rewind_cmd = rewind_cmd + " > {} 2>&1".format(pipes.quote(progress_file))
+        rewind_cmd = rewind_cmd + " > {} 2>&1".format(shlex.quote(progress_file))
         self.cmdStr = rewind_cmd
 
         Command.__init__(self, name, self.cmdStr, LOCAL)
@@ -257,7 +257,7 @@
         cmd_tokens.append('--verbose')
 
         if progress_file:
-            cmd_tokens.append('> %s 2>&1' % pipes.quote(progress_file))
+            cmd_tokens.append('> %s 2>&1' % shlex.quote(progress_file))
 
         cmd_str = ' '.join(cmd_tokens)
 
--- a/gpMgmt/bin/gppylib/commands/gp.py
+++ b/gpMgmt/bin/gppylib/commands/gp.py
@@ -12,7 +12,6 @@
 import pickle
 import shlex
 import os.path
-import pipes
 import subprocess
 
 import re, socket
@@ -939,7 +938,7 @@
                  batchSize=None, verbose=False,ctxt=LOCAL, remoteHost=None, validationOnly=False, writeGpIdFileOnly=False,
                  forceoverwrite=False):
 
-        cmdStr = '$GPHOME/bin/lib/gpconfigurenewsegment -c \"%s\" -l %s' % (confinfo, pipes.quote(logdir))
+        cmdStr = '$GPHOME/bin/lib/gpconfigurenewsegment -c \"%s\" -l %s' % (confinfo, shlex.quote(logdir))
 
         if newSegments:
             cmdStr += ' -n'
@@ -1041,7 +1040,7 @@
 
 
 def _get_cmd_for_recovery_wrapper(wrapper_filename, confinfo, logdir, batchSize, verbose, forceoverwrite, era=None):
-    cmdStr = '$GPHOME/sbin/{}.py -c {} -l {}'.format(wrapper_filename, pipes.quote(confinfo), pipes.quote(logdir))
+    cmdStr = '$GPHOME/sbin/{}.py -c {} -l {}'.format(wrapper_filename, shlex.quote(confinfo), shlex.quote(logdir))
 
     if verbose:
         cmdStr += ' -v'
--- a/gpcontrib/gp_replica_check/gp_replica_check.py
+++ b/gpcontrib/gp_replica_check/gp_replica_check.py
@@ -36,7 +36,7 @@
 except:
     import subprocess
 import threading
-import pipes  # for shell-quoting, pipes.quote()
+import shlex  # for shell-quoting, shlex.quote()
 import time
 import os
 from collections import defaultdict
@@ -78,7 +78,7 @@
                                           self.ploc, self.mloc)
 
     def wait_for_wal_sync(self):
-        cmd = "PGOPTIONS='-c gp_role=utility' psql -h %s -p %s -d %s -t -A -c \"SELECT pg_current_wal_lsn() AS master_wal, replay_lsn AS standby_wal, pg_current_wal_lsn() = replay_lsn AS are_equal FROM pg_stat_replication;\"" % (self.host, self.port, pipes.quote(self.datname))
+        cmd = "PGOPTIONS='-c gp_role=utility' psql -h %s -p %s -d %s -t -A -c \"SELECT pg_current_wal_lsn() AS master_wal, replay_lsn AS standby_wal, pg_current_wal_lsn() = replay_lsn AS are_equal FROM pg_stat_replication;\"" % (self.host, self.port, shlex.quote(self.datname))
         while True:
             try:
                 output = subprocess.check_output(cmd, shell=True).decode().strip().split("\n")
