--- cmake/CMakeLists.txt.old	2024-07-13 14:20:36.763661886 +0300
+++ cmake/CMakeLists.txt	2024-07-13 14:21:27.412461900 +0300
@@ -6,13 +6,8 @@
 # ${INSTALL_CMAKE_DIR} and @PROJECT_ROOT_DIR@ is the relative
 # path to the root from there.  (Note that the whole install tree can
 # be relocated.)
-if (NOT WIN32)
-  set (INSTALL_CMAKE_DIR "share/cmake/${PROJECT_NAME}")
-  set (PROJECT_ROOT_DIR "../../..")
-else ()
-  set (INSTALL_CMAKE_DIR "cmake")
-  set (PROJECT_ROOT_DIR "..")
-endif ()
+set (INSTALL_CMAKE_DIR "lib/cmake/${PROJECT_NAME}")
+set (PROJECT_ROOT_DIR "../../..")
 
 configure_file (project-config.cmake.in project-config.cmake @ONLY)
 configure_file (project-config-version.cmake.in
