--- a/libx509pq.c
+++ b/libx509pq.c
@@ -103,12 +103,17 @@
 #if OPENSSL_VERSION_NUMBER < 0x30000000L
 	/* We need MD2 to verify old MD2/RSA certificate signatures, but
 	  OpenSSL_add_all_digests() no longer enables MD2 by default */
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+	OPENSSL_init_crypto(OPENSSL_INIT_ADD_ALL_DIGESTS |
+		OPENSSL_INIT_LOAD_CRYPTO_STRINGS, NULL);
+#else
 	OpenSSL_add_all_digests();
+	ERR_load_crypto_strings();
+#endif
 #ifndef OPENSSL_NO_MD2
 	EVP_add_digest(EVP_md2());
 #endif
 
-	ERR_load_crypto_strings();
 #endif
 
 	/* Define the OID for the draft Basic Constraints extension. The
@@ -157,10 +162,12 @@
 		ENGINE_finish(g_gostEngine);
 		ENGINE_free(g_gostEngine);
 	}
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
 	ENGINE_cleanup();
 	EVP_cleanup();
 	OBJ_cleanup();
 	ERR_free_strings();
+#endif
 #endif
 
 	rocacheck_cleanup();
@@ -604,6 +611,10 @@
  ******************************************************************************/
 PG_FN(x509pq_opensslversion)
 {
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+	const char* t_version = OpenSSL_version(OPENSSL_VERSION);
+#else
 	const char* t_version = SSLeay_version(SSLEAY_VERSION);
+#endif
 	PG_RETURN_TEXT_P(text_from_cstring_len(t_version, strlen(t_version)));
 }
--- a/libx509pq.h
+++ b/libx509pq.h
@@ -31,6 +31,7 @@
 #endif
 #include "openssl/err.h"
 #include "openssl/evp.h"
+#include "openssl/rsa.h"
 #include "openssl/objects.h"
 #include "openssl/ocsp.h"
 #include "openssl/x509.h"
