--- a/postgis/liblwgeom/liblwgeom.h.in
+++ b/postgis/liblwgeom/liblwgeom.h.in
@@ -34,7 +34,7 @@
 #include <stdio.h>
 #include <stdbool.h>
 
-#include "../postgis_config.h"
+#include "postgis_config.h"
 
 #include "proj.h"
 
--- a/postgis/liblwgeom/liblwgeom_internal.h
+++ b/postgis/liblwgeom/liblwgeom_internal.h
@@ -29,7 +29,7 @@
 #ifndef _LIBLWGEOM_INTERNAL_H
 #define _LIBLWGEOM_INTERNAL_H 1
 
-#include "../postgis_config.h"
+#include "postgis_config.h"
 
 #include "lwgeom_log.h"
 
--- a/postgis/liblwgeom/lwgeom_geos_clean.c
+++ b/postgis/liblwgeom/lwgeom_geos_clean.c
@@ -22,7 +22,7 @@
  *
  **********************************************************************/
 
-#include "../postgis_config.h"
+#include "postgis_config.h"
 
 #include "liblwgeom.h"
 #include "lwgeom_geos.h"
--- a/postgis/liblwgeom/lwgeom_geos_split.c
+++ b/postgis/liblwgeom/lwgeom_geos_split.c
@@ -22,7 +22,7 @@
  *
  **********************************************************************/
 
-#include "../postgis_config.h"
+#include "postgis_config.h"
 /*#define POSTGIS_DEBUG_LEVEL 4*/
 #include "lwgeom_geos.h"
 #include "liblwgeom_internal.h"
--- a/postgis/liblwgeom/lwgeom_transform.c
+++ b/postgis/liblwgeom/lwgeom_transform.c
@@ -23,7 +23,7 @@
  **********************************************************************/
 
 
-#include "../postgis_config.h"
+#include "postgis_config.h"
 #include "liblwgeom_internal.h"
 #include "lwgeom_log.h"
 #include <string.h>
--- a/postgis/liblwgeom/lwgeom_wrapx.c
+++ b/postgis/liblwgeom/lwgeom_wrapx.c
@@ -22,7 +22,7 @@
  *
  **********************************************************************/
 
-#include "../postgis_config.h"
+#include "postgis_config.h"
 /*#define POSTGIS_DEBUG_LEVEL 4*/
 #include "lwgeom_geos.h"
 #include "liblwgeom_internal.h"
--- a/postgis/liblwgeom/lwin_encoded_polyline.c
+++ b/postgis/liblwgeom/lwin_encoded_polyline.c
@@ -28,7 +28,7 @@
 #include <math.h>
 
 #include "liblwgeom.h"
-#include "../postgis_config.h"
+#include "postgis_config.h"
 
 LWGEOM*
 lwgeom_from_encoded_polyline(const char *encodedpolyline, int precision)
--- a/postgis/liblwgeom/lwin_geojson.c
+++ b/postgis/liblwgeom/lwin_geojson.c
@@ -26,7 +26,7 @@
 
 #include "liblwgeom.h"
 #include "lwgeom_log.h"
-#include "../postgis_config.h"
+#include "postgis_config.h"
 
 #if defined(HAVE_LIBJSON)
 
--- a/postgis/liblwgeom/lwin_wkb.c
+++ b/postgis/liblwgeom/lwin_wkb.c
@@ -23,7 +23,7 @@
  **********************************************************************/
 
 
-#include "../postgis_config.h"
+#include "postgis_config.h"
 /*#define POSTGIS_DEBUG_LEVEL 4*/
 #include "liblwgeom_internal.h" /* NOTE: includes lwgeom_log.h */
 #include "lwgeom_log.h"
--- a/postgis/liblwgeom/lwpoint.c
+++ b/postgis/liblwgeom/lwpoint.c
@@ -26,7 +26,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include "../postgis_config.h"
+#include "postgis_config.h"
 /*#define POSTGIS_DEBUG_LEVEL 4*/
 #include "liblwgeom_internal.h"
 #include "lwgeom_log.h"
--- a/postgis/liblwgeom/lwstroke.c
+++ b/postgis/liblwgeom/lwstroke.c
@@ -30,7 +30,7 @@
 #include <stdarg.h>
 #include <string.h>
 
-#include "../postgis_config.h"
+#include "postgis_config.h"
 
 /*#define POSTGIS_DEBUG_LEVEL 3*/
 
--- a/postgis/liblwgeom/lwutil.c
+++ b/postgis/liblwgeom/lwutil.c
@@ -33,7 +33,7 @@
 #include <ctype.h> /* for tolower */
 
 /* Global variables */
-#include "../postgis_config.h"
+#include "postgis_config.h"
 #include "liblwgeom_internal.h"
 #include "lwgeom_log.h"
 
--- a/postgis/liblwgeom/ptarray.c
+++ b/postgis/liblwgeom/ptarray.c
@@ -28,7 +28,7 @@
 #include <string.h>
 #include <stdlib.h>
 
-#include "../postgis_config.h"
+#include "postgis_config.h"
 /*#define POSTGIS_DEBUG_LEVEL 4*/
 #include "liblwgeom_internal.h"
 #include "lwgeom_log.h"
--- a/postgis/libpgcommon/gserialized_gist.c
+++ b/postgis/libpgcommon/gserialized_gist.c
@@ -14,7 +14,7 @@
 #include "access/itup.h"
 #include "access/skey.h"
 
-#include "../postgis_config.h"
+#include "postgis_config.h"
 
 /*#define POSTGIS_DEBUG_LEVEL 4*/
 
--- a/postgis/libpgcommon/lwgeom_cache.c
+++ b/postgis/libpgcommon/lwgeom_cache.c
@@ -17,7 +17,7 @@
 #include "fmgr.h"
 #include "utils/memutils.h"
 
-#include "../postgis_config.h"
+#include "postgis_config.h"
 
 /* Include for VARATT_EXTERNAL_GET_POINTER */
 #if POSTGIS_PGSQL_VERSION < 130
--- a/postgis/libpgcommon/lwgeom_pg.c
+++ b/postgis/libpgcommon/lwgeom_pg.c
@@ -33,7 +33,7 @@
 #include <catalog/pg_extension.h>
 #include <commands/extension.h>
 
-#include "../postgis_config.h"
+#include "postgis_config.h"
 
 #include <utils/regproc.h>
 
--- a/postgis/libpgcommon/lwgeom_transform.c
+++ b/postgis/libpgcommon/lwgeom_transform.c
@@ -21,7 +21,7 @@
 #include "utils/hsearch.h"
 
 /* PostGIS headers */
-#include "../postgis_config.h"
+#include "postgis_config.h"
 #include "liblwgeom.h"
 #include "lwgeom_pg.h"
 #include "lwgeom_cache.h"

--- a/mobilitydb/datagen/CMakeLists.txt
+++ b/mobilitydb/datagen/CMakeLists.txt
@@ -2,7 +2,7 @@
 configure_file(mobilitydb_datagen.in.control ${CMAKE_BINARY_DIR}/mobilitydb_datagen.control)
 
 # Configure the header template directly into the extension file
-string(TIMESTAMP GENERATION_DATE "%Y-%m-%d %H:%M:%S")
+string(TIMESTAMP GENERATION_DATE "%Y-%m-%d %H:%M:%S" UTC)
 configure_file(
     header.in.sql
     ${CMAKE_BINARY_DIR}/${MOBILITYDB_DATAGEN_EXTENSION_FILE}

Description: Strip patch word from GEOS version.
 -- GEOS_VERSION: 3.15.0beta1
 -- GEOS_VERSION_MAJOR: 3beta1
 -- GEOS_VERSION_MINOR: 15beta1
 -- GEOS_VERSION_MICRO: 0beta1
 -- POSTGIS_GEOS_VERSION: ERROR
 CMake Error at CMakeLists.txt:258 (math):
   math cannot parse the expression: "3beta1 * 10000 + 15beta1 * 100 +
   0beta1": syntax error, unexpected exp_NUMBER, expecting end of file (7).
Author: Bas Couwenberg <sebastic@debian.org>

--- a/cmake/FindGEOS.cmake
+++ b/cmake/FindGEOS.cmake
@@ -72,9 +72,9 @@ ELSE(WIN32)
           IF (NOT GEOS_VERSION)
             MESSAGE (FATAL_ERROR "Could not determine GEOS version from framework.")
           ENDIF (NOT GEOS_VERSION)
-          STRING(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)(dev)?" "\\1" GEOS_VERSION_MAJOR "${GEOS_VERSION}")
-          STRING(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)(dev)?" "\\2" GEOS_VERSION_MINOR "${GEOS_VERSION}")
-          STRING(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)(dev)?" "\\3" GEOS_VERSION_MICRO "${GEOS_VERSION}")
+          STRING(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)(dev|(alpha|beta|rc)[0-9]*)?" "\\1" GEOS_VERSION_MAJOR "${GEOS_VERSION}")
+          STRING(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)(dev|(alpha|beta|rc)[0-9]*)?" "\\2" GEOS_VERSION_MINOR "${GEOS_VERSION}")
+          STRING(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)(dev|(alpha|beta|rc)[0-9]*)?" "\\3" GEOS_VERSION_MICRO "${GEOS_VERSION}")
           IF (GEOS_VERSION_MAJOR LESS 3)
             MESSAGE (FATAL_ERROR "GEOS version is too old (${GEOS_VERSION}). Use 3.0.0 or higher.")
           ENDIF (GEOS_VERSION_MAJOR LESS 3)
@@ -104,9 +104,9 @@ ELSE(WIN32)
             OUTPUT_VARIABLE GEOS_VERSION)
         # Remove trailing newline
         STRING(REGEX REPLACE "\n$" "" GEOS_VERSION "${GEOS_VERSION}")
-        STRING(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)(dev)?" "\\1" GEOS_VERSION_MAJOR "${GEOS_VERSION}")
-        STRING(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)(dev)?" "\\2" GEOS_VERSION_MINOR "${GEOS_VERSION}")
-        STRING(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)(dev)?" "\\3" GEOS_VERSION_MICRO "${GEOS_VERSION}")
+        STRING(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)(dev|(alpha|beta|rc)[0-9]*)?" "\\1" GEOS_VERSION_MAJOR "${GEOS_VERSION}")
+        STRING(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)(dev|(alpha|beta|rc)[0-9]*)?" "\\2" GEOS_VERSION_MINOR "${GEOS_VERSION}")
+        STRING(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)(dev|(alpha|beta|rc)[0-9]*)?" "\\3" GEOS_VERSION_MICRO "${GEOS_VERSION}")
 
         IF (GEOS_VERSION_MAJOR LESS 3 OR (GEOS_VERSION_MAJOR EQUAL 3 AND GEOS_VERSION_MINOR LESS 3) )
           MESSAGE (FATAL_ERROR "GEOS version is too old (${GEOS_VERSION}). Use 3.3.0 or higher.")
@@ -187,9 +187,9 @@ IF(GEOS_INCLUDE_DIR AND NOT GEOS_VERSION
   FILE(READ ${GEOS_INCLUDE_DIR}/geos_c.h VERSIONFILE)
   STRING(REGEX MATCH "#define GEOS_VERSION \"[0-9]+\\.[0-9]+\\.[0-9]+" GEOS_VERSION ${VERSIONFILE})
   STRING(REGEX MATCH "[0-9]+\\.[0-9]+\\.[0-9]+" GEOS_VERSION ${GEOS_VERSION})
-  STRING(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)(dev)?" "\\1" GEOS_VERSION_MAJOR "${GEOS_VERSION}")
-  STRING(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)(dev)?" "\\2" GEOS_VERSION_MINOR "${GEOS_VERSION}")
-  STRING(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)(dev)?" "\\3" GEOS_VERSION_MICRO "${GEOS_VERSION}")
+  STRING(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)(dev|(alpha|beta|rc)[0-9]*)?" "\\1" GEOS_VERSION_MAJOR "${GEOS_VERSION}")
+  STRING(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)(dev|(alpha|beta|rc)[0-9]*)?" "\\2" GEOS_VERSION_MINOR "${GEOS_VERSION}")
+  STRING(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)(dev|(alpha|beta|rc)[0-9]*)?" "\\3" GEOS_VERSION_MICRO "${GEOS_VERSION}")
 ENDIF(GEOS_INCLUDE_DIR AND NOT GEOS_VERSION)
 
 IF (GEOS_INCLUDE_DIR AND GEOS_LIBRARY)
