diff --git a/nominatim_fdw.c b/nominatim_fdw.c
--- a/nominatim_fdw.c
+++ b/nominatim_fdw.c
@@ -529,8 +529,13 @@
 		appendStringInfo(&buffer, "zlib %s,", ver->libz_version);
 	if (ver->libssh_version)
 		appendStringInfo(&buffer, "libSSH %s,", ver->libssh_version);
+#if LIBCURL_VERSION_NUM >= 0x074200
 	if (ver->nghttp2_version)
 		appendStringInfo(&buffer, "nghttp2 %s,", ver->nghttp2_version);
+#elif defined(CURL_VERSION_HTTP2)
+	if (ver->features & CURL_VERSION_HTTP2)
+		appendStringInfoString(&buffer, "HTTP2,");
+#endif
 
 #ifdef NOMINATIM_FDW_CC
 	appendStringInfo(&buffer, "compiled by %s,", NOMINATIM_FDW_CC);
