--- a/Cargo.toml
+++ b/Cargo.toml
@@ -18,10 +18,9 @@
 http-allow-all = []
 http-allow-azure-domains = []
 http-allow-test-domains = ["http-allow-azure-domains"]
-pg13 = ["pgrx/pg13", "pgrx-tests/pg13" ]
-pg14 = ["pgrx/pg14", "pgrx-tests/pg14" ]
-pg15 = ["pgrx/pg15", "pgrx-tests/pg15" ]
-pg16 = ["pgrx/pg16", "pgrx-tests/pg16" ]
-pg17 = ["pgrx/pg17", "pgrx-tests/pg17" ]
-pg18 = ["pgrx/pg18", "pgrx-tests/pg18" ]
+pg14 = ["pgrx/pg14" ]
+pg15 = ["pgrx/pg15" ]
+pg16 = ["pgrx/pg16" ]
+pg17 = ["pgrx/pg17" ]
+pg18 = ["pgrx/pg18" ]
 unsafe = ["pgrx/unsafe-postgres"]
@@ -30,4 +29,4 @@
 [dependencies]
-pgrx = "=0.16.1"
+pgrx = "=0.18.1"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = { version = "1.0", features = ["preserve_order"] }
@@ -55,4 +54,1 @@
-[dev-dependencies]
-pgrx-tests = "=0.16.1"
-
 [profile.dev]
