--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,11 +23,12 @@
 pg_test = ["extension", "dep:pgrx-tests", "pgrx-tests/pg18"]
 
 [dependencies]
-pgrx = {version = "=0.17.0", optional = true}
-pgrx-tests = {version = "=0.17.0", optional = true}
+pgrx = {version = "=0.18.1", optional = true}
+pgrx-tests = {version = "=0.18.1", optional = true}
 
 [build-dependencies]
 cc = "1.2"
+pkg-config = "0.3"
 
 [profile.dev]
 panic = "unwind"
--- a/build.rs
+++ b/build.rs
@@ -5,11 +5,16 @@
         return;
     }
 
-    cc::Build::new()
-        .file("src/rtgeom_fast.c")
+    let rttopo = pkg_config::Config::new()
+        .probe("rttopo")
+        .expect("librttopo-dev with rttopo.pc is required to build pg_eviltransform");
+
+    let mut build = cc::Build::new();
+    for include_path in rttopo.include_paths {
+        build.include(include_path);
+    }
+    build.file("src/rtgeom_fast.c")
         .flag_if_supported("-O3")
         .compile("pgct_rtgeom_fast");
-
-    println!("cargo:rustc-link-lib=rttopo");
     println!("cargo:rustc-link-lib=m");
 }
