diff --git a/Cargo.toml b/Cargo.toml
index 50dd451..e1ef49d 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,8 +32,8 @@ pg_test = ["extension", "dep:pgrx-tests"]
 
 [dependencies]
 natural-breaks = { version = "0.2.0", features = ["low-memory"] }
-pgrx = {version = "=0.19.1", optional = true}
-pgrx-tests = {version = "=0.19.1", optional = true}
+pgrx = { version = "=0.19.1", optional = true }
+pgrx-tests = { version = "=0.19.1", optional = true }
 
 [build-dependencies]
 cc = "1.2"
