--- a/src/binary/connection.c
+++ b/src/binary/connection.c
@@ -123,15 +123,17 @@ static void
 tls_connect(struct ch_binary_state *s, const char *host)
 {
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
 	static int	openssl_init_done = 0;
 
 	if (!openssl_init_done)
 	{
 		SSL_library_init();
 		SSL_load_error_strings();
 		OpenSSL_add_all_algorithms();
 		openssl_init_done = 1;
 	}
+#endif
 
 	s->ssl_ctx = SSL_CTX_new(TLS_client_method());
 	if (!s->ssl_ctx)
 		ereport(ERROR,
 				(errcode(ERRCODE_FDW_UNABLE_TO_ESTABLISH_CONNECTION),
