--- a/src/binary/connection.c
+++ b/src/binary/connection.c
@@ -169,6 +169,7 @@ openssl_min_proto_version(tls_version v) {
 
 static void
 tls_connect(struct ch_binary_state* s, const char* host, tls_version min_version) {
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
     static int openssl_init_done = 0;
 
     if (!openssl_init_done) {
@@ -177,6 +178,7 @@ tls_connect(struct ch_binary_state* s, const char* host, tls_version min_version
         OpenSSL_add_all_algorithms();
         openssl_init_done = 1;
     }
+#endif
 
     s->ssl_ctx = SSL_CTX_new(TLS_client_method());
     if (!s->ssl_ctx) {
