--- a/Cargo.toml	2026-06-08 13:04:21.000000000 +0800
+++ b/Cargo.toml	2026-06-11 21:40:00.000000000 +0800
@@ -20,8 +20,8 @@
 pg_test = ["extension", "dep:pgrx-tests", "pgrx-tests/pg18"]

 [dependencies]
-pgrx = { version = "=0.17.0", optional = true }
-pgrx-tests = { version = "=0.17.0", optional = true }
+pgrx = { version = "=0.18.1", optional = true }
+pgrx-tests = { version = "=0.18.1", optional = true }
 serde_json = { version = "1.0", optional = true }

 [profile.dev]
