diff -ruN a/LICENSE b/LICENSE
--- a/LICENSE	2026-09-01 07:12:37
+++ b/LICENSE	2026-09-01 07:12:37
@@ -69,4 +69,23 @@
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-SOFTWARE.
\ No newline at end of file
+SOFTWARE.
+
+# libsodium License
+
+ISC License
+
+Copyright (c) 2013-2026
+Frank Denis <j at pureftpd dot org>
+
+Permission to use, copy, modify, and/or distribute this software for any
+purpose with or without fee is hereby granted, provided that the above
+copyright notice and this permission notice appear in all copies.
+
+THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
diff -ruN a/Makefile b/Makefile
--- a/Makefile	2026-09-01 07:12:37
+++ b/Makefile	2026-09-01 07:12:37
@@ -6,8 +6,11 @@
 PGXS := $(shell $(PG_CONFIG) --pgxs)
 MODULE_big = pgsodium
 OBJS = $(patsubst %.c,%.o,$(wildcard src/*.c))
-SHLIB_LINK = -lsodium
-PG_CPPFLAGS = -O0
+PGSODIUM_LIBSODIUM_INCLUDEDIR ?= /usr/include/pgsodium-libsodium
+PGSODIUM_LIBSODIUM_LIBDIR ?= /usr/lib/pgsodium-libsodium
+SHLIB_LINK = $(PGSODIUM_LIBSODIUM_LIBDIR)/libsodium-pgsodium.a
+PG_CPPFLAGS = -O0 -I$(PGSODIUM_LIBSODIUM_INCLUDEDIR) \
+	-include $(PGSODIUM_LIBSODIUM_INCLUDEDIR)/pgsodium-libsodium-symbols.h
 
 TESTS        = $(wildcard sql/*.sql)
 REGRESS      = $(patsubst sql/%.sql,%,$(TESTS))
