--- a/pgspider_ext.c
+++ b/pgspider_ext.c
@@ -22,6 +22,9 @@
 #include "access/table.h"
 #include "catalog/namespace.h"
 #include "commands/explain.h"
+#if PG_VERSION_NUM >= 180000
+#include "commands/explain_state.h"
+#endif
 #include "catalog/partition.h"
 #include "foreign/fdwapi.h"
 #include "foreign/foreign.h"
@@ -876,6 +879,9 @@
 		add_path(baserel, (Path *) create_foreignscan_path(root, baserel,
 														   target,
 														   rows,
+#if PG_VERSION_NUM >= 180000
+														   childpath->disabled_nodes,
+#endif
 														   startup_cost,
 														   total_cost,
 														   pathkeys,
@@ -1280,7 +1286,9 @@
 	 * Init range table, in which we use range table array for exec_rt_fetch()
 	 * because it is faster than rt_fetch().
 	 */
-#if PG_VERSION_NUM >= 160000
+#if PG_VERSION_NUM >= 180000
+	ExecInitRangeTable(child_estate, query->rtable, permInfos, NULL);
+#elif PG_VERSION_NUM >= 160000
 	ExecInitRangeTable(child_estate, query->rtable, permInfos);
 #else
 	ExecInitRangeTable(child_estate, query->rtable);
@@ -2024,6 +2032,9 @@
 										  grouped_rel,
 										  grouped_rel->reltarget,
 										  rows,
+#if PG_VERSION_NUM >= 180000
+										  child_path->disabled_nodes,
+#endif
 										  startup_cost,
 										  total_cost,
 										  NIL,	/* no pathkeys */
