diff --git a/include/sequence/decompression_iteration.h b/include/sequence/decompression_iteration.h
index 85cd934..d1e971e 100644
--- a/include/sequence/decompression_iteration.h
+++ b/include/sequence/decompression_iteration.h
@@ -16,7 +16,7 @@
 #define SEQUENCE_DECOMPRESSION_ITERATION_H_
 
 #include "postgres.h"
-#include "access/tuptoaster.h"
+#include "access/detoast.h"
 #include "c.h"
 
 #include "sequence/sequence.h"
@@ -155,7 +155,7 @@ static PB_DecodingMap* get_decoding_map(const PB_CodeSet* codeset, int mode)
 \
 	PB_TRACE(errmsg("BEGIN_DECODE(%u,%u)", __pb_decode_start_position, __pb_decode_output_length));\
 \
-	__pb_decode_input_header = (PB_CompressedSequence*) PG_DETOAST_DATUM_SLICE(__pb_decode_input,\
+	__pb_decode_input_header = (PB_CompressedSequence*) PG_DETOAST_DATUM_SLICE(PointerGetDatum(__pb_decode_input),\
 																			   0,\
 																			   sizeof(PB_CompressedSequence) - VARHDRSZ);\
 \
@@ -178,7 +178,7 @@ static PB_DecodingMap* get_decoding_map(const PB_CodeSet* codeset, int mode)
 \
 		pfree(__pb_decode_input_header);\
 		__pb_decode_input_header = (PB_CompressedSequence*)\
-									PG_DETOAST_DATUM_SLICE(__pb_decode_input,0,sizeof(PB_CompressedSequence) - VARHDRSZ +\
+									PG_DETOAST_DATUM_SLICE(PointerGetDatum(__pb_decode_input),0,sizeof(PB_CompressedSequence) - VARHDRSZ +\
 														   __pb_decode_code_size);\
 \
 		__pb_decode_code = PB_COMPRESSED_SEQUENCE_SYMBOL_POINTER(__pb_decode_input_header);\
@@ -205,7 +205,7 @@ static PB_DecodingMap* get_decoding_map(const PB_CodeSet* codeset, int mode)
 		__pb_decode_start_entry_no = (__pb_decode_start_position + 1) / PB_INDEX_PART_SIZE - 1;\
 		if (__pb_decode_start_entry_no >= 0) {\
 			Varlena* __pb_decode_data_slice = (Varlena*)\
-				PG_DETOAST_DATUM_SLICE(__pb_decode_input,\
+				PG_DETOAST_DATUM_SLICE(PointerGetDatum(__pb_decode_input),\
 									   sizeof(PB_CompressedSequence) - VARHDRSZ +\
 									   sizeof(PB_Codeword) * __pb_decode_input_header->n_symbols +\
 									   sizeof(PB_IndexEntry) * __pb_decode_start_entry_no,\
@@ -237,7 +237,7 @@ static PB_DecodingMap* get_decoding_map(const PB_CodeSet* codeset, int mode)
 			__pb_decode_slice_size = __pb_decode_raw_size - __pb_decode_stream_offset;\
 \
 		__pb_decode_input_slice = (Varlena*)\
-				PG_DETOAST_DATUM_SLICE(__pb_decode_input,\
+				PG_DETOAST_DATUM_SLICE(PointerGetDatum(__pb_decode_input),\
 									   __pb_decode_stream_offset,\
 									   __pb_decode_slice_size);\
 \
@@ -263,7 +263,7 @@ static PB_DecodingMap* get_decoding_map(const PB_CodeSet* codeset, int mode)
 		int __pb_decode_slice_start = __pb_decode_stream_offset +\
 									  __pb_decode_start_entry->block * PB_COMPRESSION_BUFFER_BYTE_SIZE;\
 		int __pb_decode_slice_size = (__pb_decode_output_length + (__pb_decode_start_position % PB_INDEX_PART_SIZE)) *\
-									  __pb_decode_max_codeword_length;\
+									  __pb_decode_max_codeword_length + __pb_decode_start_entry->bit;\
 \
 		if (__pb_decode_codeset->uses_rle)\
 			__pb_decode_slice_size += __pb_decode_max_codeword_length + 8;\
@@ -275,7 +275,7 @@ static PB_DecodingMap* get_decoding_map(const PB_CodeSet* codeset, int mode)
 			__pb_decode_slice_size = __pb_decode_raw_size - __pb_decode_slice_start;\
 \
 		__pb_decode_input_slice = (Varlena*)\
-					PG_DETOAST_DATUM_SLICE(__pb_decode_input,\
+					PG_DETOAST_DATUM_SLICE(PointerGetDatum(__pb_decode_input),\
 										   __pb_decode_slice_start,\
 										   __pb_decode_slice_size);\
 \
diff --git a/include/sequence/sequence.h b/include/sequence/sequence.h
index aff8650..a71aba3 100644
--- a/include/sequence/sequence.h
+++ b/include/sequence/sequence.h
@@ -15,6 +15,9 @@
 #define SEQUENCE_SEQUENCE_H_
 
 #include "postgres.h"
+#if PG_VERSION_NUM >= 160000
+#include "varatt.h"
+#endif
 
 /**
  * Size of input string alphabet
diff --git a/src/sequence/compression.c b/src/sequence/compression.c
index cff3b6b..aa3bf75 100644
--- a/src/sequence/compression.c
+++ b/src/sequence/compression.c
@@ -13,7 +13,7 @@
 */
 #include "postgres.h"
 #include "fmgr.h"
-#include "access/tuptoaster.h"
+#include "access/detoast.h"
 #include "c.h"
 
 #include "sequence/sequence.h"
@@ -1864,7 +1864,7 @@ static void decode_pc_idx(Varlena* input,
 	 * Calculate the offset of the stream in the compressed data.
 	 */
 	sequence_header = (PB_CompressedSequence*)
-					  PG_DETOAST_DATUM_SLICE(input, 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
+					  PG_DETOAST_DATUM_SLICE(PointerGetDatum(input), 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
 	stream_offset = PB_COMPRESSED_SEQUENCE_STREAM_OFFSET(sequence_header) - VARHDRSZ;
 	pfree(sequence_header);
 
@@ -1889,7 +1889,7 @@ static void decode_pc_idx(Varlena* input,
 		slice_start += stream_offset;
 		slice_size *= PB_COMPRESSION_BUFFER_BYTE_SIZE;
 		input_slice = (Varlena*)
-					  PG_DETOAST_DATUM_SLICE(input,
+					  PG_DETOAST_DATUM_SLICE(PointerGetDatum(input),
 											 slice_start,
 											 slice_size);
 
@@ -1927,7 +1927,7 @@ static void decode_pc_idx(Varlena* input,
 				slice_size = raw_size - stream_offset;
 
 			input_slice = (Varlena*)
-						  PG_DETOAST_DATUM_SLICE(input,
+						  PG_DETOAST_DATUM_SLICE(PointerGetDatum(input),
 												 stream_offset,
 												 slice_size);
 
@@ -1947,7 +1947,7 @@ static void decode_pc_idx(Varlena* input,
 			int slice_start = stream_offset +
 							  start_entry->block * PB_COMPRESSION_BUFFER_BYTE_SIZE;
-			int slice_size = (output_length + (start_position % PB_INDEX_PART_SIZE)) *
-							 max_codeword_length /
+			int slice_size = ((output_length + (start_position % PB_INDEX_PART_SIZE)) *
+							 max_codeword_length + start_entry->bit) /
 							 PB_COMPRESSION_BUFFER_BIT_SIZE + 1;
 			slice_size *= PB_COMPRESSION_BUFFER_BYTE_SIZE;
 
@@ -1958,7 +1958,7 @@ static void decode_pc_idx(Varlena* input,
 				slice_size = raw_size - slice_start;
 
 			input_slice = (Varlena*)
-						  PG_DETOAST_DATUM_SLICE(input,
+						  PG_DETOAST_DATUM_SLICE(PointerGetDatum(input),
 												 slice_start,
 												 slice_size);
 
@@ -2040,7 +2040,7 @@ static void decode_pc_rle_idx(Varlena* input,
 	 * Calculate the offset of the stream in the compressed data.
 	 */
 	sequence_header = (PB_CompressedSequence*)
-					  PG_DETOAST_DATUM_SLICE(input, 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
+					  PG_DETOAST_DATUM_SLICE(PointerGetDatum(input), 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
 	stream_offset = PB_COMPRESSED_SEQUENCE_STREAM_OFFSET(sequence_header) - VARHDRSZ;
 	pfree(sequence_header);
 
@@ -2066,7 +2066,7 @@ static void decode_pc_rle_idx(Varlena* input,
 			slice_size = raw_size - stream_offset;
 
 		input_slice = (Varlena*)
-					  PG_DETOAST_DATUM_SLICE(input,
+					  PG_DETOAST_DATUM_SLICE(PointerGetDatum(input),
 											 stream_offset,
 											 slice_size);
 
@@ -2086,7 +2086,7 @@ static void decode_pc_rle_idx(Varlena* input,
 		int slice_start = stream_offset +
 						  start_entry->block * PB_COMPRESSION_BUFFER_BYTE_SIZE;
 		int slice_size = ((output_length + (start_position % PB_INDEX_PART_SIZE) + 1) *
-						 max_codeword_length  + 8) /
+						 max_codeword_length + start_entry->bit + 8) /
 						 PB_COMPRESSION_BUFFER_BIT_SIZE + 1;
 		slice_size *= PB_COMPRESSION_BUFFER_BYTE_SIZE;
 
@@ -2097,7 +2097,7 @@ static void decode_pc_rle_idx(Varlena* input,
 			slice_size = raw_size - slice_start;
 
 		input_slice = (Varlena*)
-					  PG_DETOAST_DATUM_SLICE(input,
+					  PG_DETOAST_DATUM_SLICE(PointerGetDatum(input),
 											 slice_start,
 											 slice_size);
 
@@ -2248,7 +2248,7 @@ static void decode_pc_swp_idx(Varlena* input,
 	 * Calculate the offset of the stream in the compressed data.
 	 */
 	sequence_header = (PB_CompressedSequence*)
-					  PG_DETOAST_DATUM_SLICE(input, 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
+					  PG_DETOAST_DATUM_SLICE(PointerGetDatum(input), 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
 	stream_offset = PB_COMPRESSED_SEQUENCE_STREAM_OFFSET(sequence_header) - VARHDRSZ;
 	pfree(sequence_header);
 
@@ -2274,7 +2274,7 @@ static void decode_pc_swp_idx(Varlena* input,
 			slice_size = raw_size - stream_offset;
 
 		input_slice = (Varlena*)
-					  PG_DETOAST_DATUM_SLICE(input,
+					  PG_DETOAST_DATUM_SLICE(PointerGetDatum(input),
 											 stream_offset,
 											 slice_size);
 
@@ -2298,7 +2298,7 @@ static void decode_pc_swp_idx(Varlena* input,
 		int slice_start = stream_offset +
 						  start_entry->block * PB_COMPRESSION_BUFFER_BYTE_SIZE;
-		int slice_size = (output_length + (start_position % PB_INDEX_PART_SIZE)) *
-						 max_codeword_length /
+		int slice_size = ((output_length + (start_position % PB_INDEX_PART_SIZE)) *
+						 max_codeword_length + start_entry->bit) /
 						 PB_COMPRESSION_BUFFER_BIT_SIZE + 1;
 		slice_size *= PB_COMPRESSION_BUFFER_BYTE_SIZE;
 
@@ -2309,7 +2309,7 @@ static void decode_pc_swp_idx(Varlena* input,
 			slice_size = raw_size - slice_start;
 
 		input_slice = (Varlena*)
-					  PG_DETOAST_DATUM_SLICE(input,
+					  PG_DETOAST_DATUM_SLICE(PointerGetDatum(input),
 											 slice_start,
 											 slice_size);
 
@@ -2434,7 +2434,7 @@ static void decode_pc_swp_rle_idx(Varlena* input,
 	 * Calculate the offset of the stream in the compressed data.
 	 */
 	sequence_header = (PB_CompressedSequence*)
-					  PG_DETOAST_DATUM_SLICE(input, 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
+					  PG_DETOAST_DATUM_SLICE(PointerGetDatum(input), 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
 	stream_offset = PB_COMPRESSED_SEQUENCE_STREAM_OFFSET(sequence_header) - VARHDRSZ;
 	pfree(sequence_header);
 
@@ -2460,7 +2460,7 @@ static void decode_pc_swp_rle_idx(Varlena* input,
 			slice_size = raw_size - stream_offset;
 
 		input_slice = (Varlena*)
-					  PG_DETOAST_DATUM_SLICE(input,
+					  PG_DETOAST_DATUM_SLICE(PointerGetDatum(input),
 											 stream_offset,
 											 slice_size);
 
@@ -2486,7 +2486,7 @@ static void decode_pc_swp_rle_idx(Varlena* input,
 		int slice_start = stream_offset +
 						  start_entry->block * PB_COMPRESSION_BUFFER_BYTE_SIZE;
-		int slice_size = (output_length + (start_position % PB_INDEX_PART_SIZE)) *
-						 max_codeword_length /
+		int slice_size = ((output_length + (start_position % PB_INDEX_PART_SIZE)) *
+						 max_codeword_length + start_entry->bit) /
 						 PB_COMPRESSION_BUFFER_BIT_SIZE + 1;
 		slice_size *= PB_COMPRESSION_BUFFER_BYTE_SIZE;
 
@@ -2497,7 +2497,7 @@ static void decode_pc_swp_rle_idx(Varlena* input,
 			slice_size = raw_size - slice_start;
 
 		input_slice = (Varlena*)
-					  PG_DETOAST_DATUM_SLICE(input,
+					  PG_DETOAST_DATUM_SLICE(PointerGetDatum(input),
 											 slice_start,
 											 slice_size);
 
@@ -2896,7 +2896,7 @@ void decode(Varlena* input,
 	/*
 	 * detoast only the header of the sequence
 	 */
-	input_header = (PB_CompressedSequence*) PG_DETOAST_DATUM_SLICE(input,
+	input_header = (PB_CompressedSequence*) PG_DETOAST_DATUM_SLICE(PointerGetDatum(input),
 																   0,
 																   sizeof(PB_CompressedSequence) - VARHDRSZ);
 
@@ -2928,7 +2928,7 @@ void decode(Varlena* input,
 
 		pfree(input_header);
 		input_header = (PB_CompressedSequence*)
-					   PG_DETOAST_DATUM_SLICE(input,0,sizeof(PB_CompressedSequence) - VARHDRSZ +
+					   PG_DETOAST_DATUM_SLICE(PointerGetDatum(input),0,sizeof(PB_CompressedSequence) - VARHDRSZ +
 													  code_size);
 
 		code = PB_COMPRESSED_SEQUENCE_SYMBOL_POINTER(input_header);
@@ -2952,7 +2952,7 @@ void decode(Varlena* input,
 		start_entry_no = (start_position + 1) / PB_INDEX_PART_SIZE - 1;
 		if (start_entry_no >= 0)
 		{
-			Varlena* data_slice = (Varlena*) PG_DETOAST_DATUM_SLICE(input,
+			Varlena* data_slice = (Varlena*) PG_DETOAST_DATUM_SLICE(PointerGetDatum(input),
 													  sizeof(PB_CompressedSequence) - VARHDRSZ +
 													  sizeof(PB_Codeword) * input_header->n_symbols +
 													  sizeof(PB_IndexEntry) * start_entry_no,
@@ -3011,4 +3011,3 @@ void decode(Varlena* input,
 
 	pfree(input_header);
 }
-
diff --git a/src/sequence/functions.c b/src/sequence/functions.c
index c622e4a..ef9502a 100644
--- a/src/sequence/functions.c
+++ b/src/sequence/functions.c
@@ -17,7 +17,7 @@
 #include "fmgr.h"
 #include "utils/array.h"
 #include "catalog/pg_type.h"
-#include "access/tuptoaster.h"
+#include "access/detoast.h"
 
 #include "sequence/sequence.h"
 #include "sequence/stats.h"
@@ -120,13 +120,13 @@ bool sequence_equal(Varlena* raw_seq1, Varlena* raw_seq2, PB_CodeSet** fixed_cod
 
 	/* Get length of first sequence */
 	input_header = (PB_CompressedSequence*)
-					PG_DETOAST_DATUM_SLICE(raw_seq1,0,4);
+					PG_DETOAST_DATUM_SLICE(PointerGetDatum(raw_seq1),0,4);
 
 	len = input_header->sequence_length;
 
 	/* Get length of second sequence */
 	input_header = (PB_CompressedSequence*)
-					PG_DETOAST_DATUM_SLICE(raw_seq2,0,4);
+					PG_DETOAST_DATUM_SLICE(PointerGetDatum(raw_seq2),0,4);
 
 	/* Compare length */
 	if (len != input_header->sequence_length)
@@ -135,8 +135,8 @@ bool sequence_equal(Varlena* raw_seq1, Varlena* raw_seq2, PB_CodeSet** fixed_cod
 		return false;
 	}
 
-	seq1 = (PB_CompressedSequence*) PG_DETOAST_DATUM(raw_seq1);
-	seq2 = (PB_CompressedSequence*) PG_DETOAST_DATUM(raw_seq2);
+	seq1 = (PB_CompressedSequence*) PG_DETOAST_DATUM(PointerGetDatum(raw_seq1));
+	seq2 = (PB_CompressedSequence*) PG_DETOAST_DATUM(PointerGetDatum(raw_seq2));
 
 	decompressed_sequence = palloc0(len);
 	decode((Varlena*) seq1, decompressed_sequence, 0, len, fixed_codesets);
@@ -169,8 +169,8 @@ bool sequence_equal(Varlena* raw_seq1, Varlena* raw_seq2, PB_CodeSet** fixed_cod
  */
 int sequence_compare(Varlena* seq_a, Varlena* seq_b, PB_CodeSet** fixed_codesets)
 {
-	PB_CompressedSequence* seq1 = (PB_CompressedSequence*) PG_DETOAST_DATUM(seq_a);
-	PB_CompressedSequence* seq2 = (PB_CompressedSequence*) PG_DETOAST_DATUM(seq_b);
+	PB_CompressedSequence* seq1 = (PB_CompressedSequence*) PG_DETOAST_DATUM(PointerGetDatum(seq_a));
+	PB_CompressedSequence* seq2 = (PB_CompressedSequence*) PG_DETOAST_DATUM(PointerGetDatum(seq_b));
 	int swap = 1;
 
 	uint8* decompressed_sequence;
diff --git a/src/sequence/generation.c b/src/sequence/generation.c
index 2723ee3..51f2a79 100644
--- a/src/sequence/generation.c
+++ b/src/sequence/generation.c
@@ -32,7 +32,7 @@ Datum generate_sequence (PG_FUNCTION_ARGS);
 PG_FUNCTION_INFO_V1 (generate_sequence);
 Datum generate_sequence (PG_FUNCTION_ARGS) {
 	PB_Alphabet* input =
-			(PB_Alphabet*) PG_DETOAST_DATUM(PG_GETARG_POINTER(0));
+			(PB_Alphabet*) PG_DETOAST_DATUM(PG_GETARG_DATUM(0));
 	int32 sequence_length = PG_GETARG_INT32(1);
 
 	text* result;
diff --git a/src/types/aa_sequence.c b/src/types/aa_sequence.c
index 9d1ccc6..8801f37 100644
--- a/src/types/aa_sequence.c
+++ b/src/types/aa_sequence.c
@@ -17,7 +17,7 @@
 #include "fmgr.h"
 #include "utils/array.h"
 #include "catalog/pg_type.h"
-#include "access/tuptoaster.h"
+#include "access/detoast.h"
 
 #include "sequence/sequence.h"
 #include "sequence/stats.h"
@@ -42,7 +42,7 @@ static PB_CodeSet aa_iupac = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) true,
-		.fixed_id = (bool) 0,
+		.fixed_id = (uint8) 0,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 0,
 		.ascii_bitmap_high = (uint64) 132086782,
@@ -69,7 +69,7 @@ static PB_CodeSet aa_iupac_cs = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) false,
-		.fixed_id = (bool) 1,
+		.fixed_id = (uint8) 1,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 0,
 		.ascii_bitmap_high = (uint64) 567308409055968254,
@@ -390,7 +390,7 @@ Datum aa_sequence_typmod_out(PG_FUNCTION_ARGS) {
 
 	PB_TRACE(errmsg("<-aa_sequence_typmod_out() returning '%s'", out));
 
-	PG_RETURN_INT32(result);
+	PG_RETURN_CSTRING(result);
 }
 
 /**
@@ -609,7 +609,7 @@ Datum aa_sequence_substring (PG_FUNCTION_ARGS) {
 	PB_TRACE(errmsg("->aa_sequence_substring()"));
 
 	input_header = (PB_CompressedSequence*)
-			PG_DETOAST_DATUM_SLICE(input, 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
+			PG_DETOAST_DATUM_SLICE(PointerGetDatum(input), 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
 
 	if (len < 0) {
 		ereport(ERROR,(errmsg("negative substring length not allowed")));
@@ -653,7 +653,7 @@ PG_FUNCTION_INFO_V1 (aa_sequence_char_length);
 Datum aa_sequence_char_length (PG_FUNCTION_ARGS)
 {
 	PB_CompressedSequence* seq = (PB_CompressedSequence*)
-			PG_DETOAST_DATUM_SLICE(PG_GETARG_RAW_VARLENA_P(0),0,4);
+			PG_DETOAST_DATUM_SLICE(PG_GETARG_DATUM(0),0,4);
 
 	PG_RETURN_INT32(seq->sequence_length);
 }
@@ -673,7 +673,7 @@ Datum aa_sequence_compression_ratio (PG_FUNCTION_ARGS)
 {
 	Varlena* input = (Varlena*) PG_GETARG_RAW_VARLENA_P(0);
 	PB_CompressedSequence* input_header = (PB_CompressedSequence*)
-		PG_DETOAST_DATUM_SLICE(input, 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
+		PG_DETOAST_DATUM_SLICE(PointerGetDatum(input), 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
 
 	double cr = (double) toast_raw_datum_size((Datum)input) / (double) (input_header->sequence_length + VARHDRSZ);
 
diff --git a/src/types/aligned_aa_sequence.c b/src/types/aligned_aa_sequence.c
index 1b00b5d..9682ef7 100644
--- a/src/types/aligned_aa_sequence.c
+++ b/src/types/aligned_aa_sequence.c
@@ -17,7 +17,7 @@
 #include "fmgr.h"
 #include "utils/array.h"
 #include "catalog/pg_type.h"
-#include "access/tuptoaster.h"
+#include "access/detoast.h"
 
 #include "sequence/sequence.h"
 #include "sequence/stats.h"
@@ -42,7 +42,7 @@ static PB_CodeSet aligned_aa_iupac = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) true,
-		.fixed_id = (bool) 0,
+		.fixed_id = (uint8) 0,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 105553116266496,
 		.ascii_bitmap_high = (uint64) 132086782,
@@ -64,7 +64,7 @@ static PB_CodeSet aligned_aa_iupac_cs = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) false,
-		.fixed_id = (bool) 1,
+		.fixed_id = (uint8) 1,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 105553116266496,
 		.ascii_bitmap_high = (uint64) 567308409055968254,
@@ -367,7 +367,7 @@ Datum aligned_aa_sequence_typmod_out(PG_FUNCTION_ARGS)
 	out++;
 	*out = 0;
 
-	PG_RETURN_INT32(result);
+	PG_RETURN_CSTRING(result);
 }
 
 /**
@@ -587,7 +587,7 @@ Datum aligned_aa_sequence_substring (PG_FUNCTION_ARGS) {
 	PB_TRACE(errmsg("->aligned_aa_sequence_substring()"));
 
 	input_header = (PB_CompressedSequence*)
-			PG_DETOAST_DATUM_SLICE(input, 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
+			PG_DETOAST_DATUM_SLICE(PointerGetDatum(input), 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
 
 	if (len < 0) {
 		ereport(ERROR,(errmsg("negative substring length not allowed")));
@@ -631,7 +631,7 @@ PG_FUNCTION_INFO_V1 (aligned_aa_sequence_char_length);
 Datum aligned_aa_sequence_char_length (PG_FUNCTION_ARGS)
 {
 	PB_CompressedSequence* seq = (PB_CompressedSequence*)
-			PG_DETOAST_DATUM_SLICE(PG_GETARG_RAW_VARLENA_P(0),0,4);
+			PG_DETOAST_DATUM_SLICE(PG_GETARG_DATUM(0),0,4);
 
 	PG_RETURN_INT32(seq->sequence_length);
 }
@@ -651,7 +651,7 @@ Datum aligned_aa_sequence_compression_ratio (PG_FUNCTION_ARGS)
 {
 	Varlena* input = (Varlena*) PG_GETARG_RAW_VARLENA_P(0);
 	PB_CompressedSequence* input_header = (PB_CompressedSequence*)
-		PG_DETOAST_DATUM_SLICE(input, 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
+		PG_DETOAST_DATUM_SLICE(PointerGetDatum(input), 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
 
 	double cr = (double) toast_raw_datum_size((Datum)input) / (double) (input_header->sequence_length + VARHDRSZ);
 
diff --git a/src/types/aligned_dna_sequence.c b/src/types/aligned_dna_sequence.c
index ca6a27d..5398c4a 100644
--- a/src/types/aligned_dna_sequence.c
+++ b/src/types/aligned_dna_sequence.c
@@ -17,7 +17,7 @@
 #include "fmgr.h"
 #include "utils/array.h"
 #include "catalog/pg_type.h"
-#include "access/tuptoaster.h"
+#include "access/detoast.h"
 
 #include "sequence/sequence.h"
 #include "sequence/stats.h"
@@ -42,7 +42,7 @@ static PB_CodeSet aligned_dna_flc = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) true,
-		.fixed_id = (bool) 0,
+		.fixed_id = (uint8) 0,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 105553116266496,
 		.ascii_bitmap_high = (uint64) 1048714,
@@ -58,7 +58,7 @@ static PB_CodeSet aligned_dna_flc_cs = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) false,
-		.fixed_id = (bool) 1,
+		.fixed_id = (uint8) 1,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 105553116266496,
 		.ascii_bitmap_high = (uint64) 4504192333906058,
@@ -75,7 +75,7 @@ static PB_CodeSet aligned_dna_iupac = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) true,
-		.fixed_id = (bool) 2,
+		.fixed_id = (uint8) 2,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 105553116266496,
 		.ascii_bitmap_high = (uint64) 47999390,
@@ -95,7 +95,7 @@ static PB_CodeSet aligned_dna_iupac_cs = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) false,
-		.fixed_id = (bool) 3,
+		.fixed_id = (uint8) 3,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 105553116266496,
 		.ascii_bitmap_high = (uint64) 206155810325948830,
@@ -119,7 +119,7 @@ static PB_CodeSet aligned_dna_flc_complement = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) true,
-		.fixed_id = (bool) 4,
+		.fixed_id = (uint8) 4,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 105553116266496,
 		.ascii_bitmap_high = (uint64) 1048714,
@@ -135,7 +135,7 @@ static PB_CodeSet aligned_dna_flc_cs_complement = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) false,
-		.fixed_id = (bool) 5,
+		.fixed_id = (uint8) 5,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 105553116266496,
 		.ascii_bitmap_high = (uint64) 4504192333906058,
@@ -152,7 +152,7 @@ static PB_CodeSet aligned_dna_iupac_complement = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) true,
-		.fixed_id = (bool) 6,
+		.fixed_id = (uint8) 6,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 105553116266496,
 		.ascii_bitmap_high = (uint64) 47999390,
@@ -172,7 +172,7 @@ static PB_CodeSet aligned_dna_iupac_cs_complement = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) false,
-		.fixed_id = (bool) 7,
+		.fixed_id = (uint8) 7,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 105553116266496,
 		.ascii_bitmap_high = (uint64) 206155810325948830,
@@ -505,7 +505,7 @@ Datum aligned_dna_sequence_typmod_out(PG_FUNCTION_ARGS) {
 
 	PB_TRACE(errmsg("<-aligned_dna_sequence_typmod_out() returning '%s'", out));
 
-	PG_RETURN_INT32(result);
+	PG_RETURN_CSTRING(result);
 }
 
 /**
@@ -725,7 +725,7 @@ Datum aligned_dna_sequence_substring (PG_FUNCTION_ARGS)
 	PB_TRACE(errmsg("->aligned_dna_sequence_substring()"));
 
 	input_header = (PB_CompressedSequence*)
-			PG_DETOAST_DATUM_SLICE(input, 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
+			PG_DETOAST_DATUM_SLICE(PointerGetDatum(input), 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
 
 	if (len < 0) {
 		ereport(ERROR,(errmsg("negative substring length not allowed")));
@@ -769,7 +769,7 @@ PG_FUNCTION_INFO_V1 (aligned_dna_sequence_char_length);
 Datum aligned_dna_sequence_char_length (PG_FUNCTION_ARGS)
 {
 	PB_CompressedSequence* seq = (PB_CompressedSequence*)
-			PG_DETOAST_DATUM_SLICE(PG_GETARG_RAW_VARLENA_P(0),0,4);
+			PG_DETOAST_DATUM_SLICE(PG_GETARG_DATUM(0),0,4);
 
 	PG_RETURN_INT32(seq->sequence_length);
 }
@@ -789,7 +789,7 @@ Datum aligned_dna_sequence_compression_ratio (PG_FUNCTION_ARGS)
 {
 	Varlena* input = (Varlena*) PG_GETARG_RAW_VARLENA_P(0);
 	PB_CompressedSequence* input_header = (PB_CompressedSequence*)
-		PG_DETOAST_DATUM_SLICE(input, 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
+		PG_DETOAST_DATUM_SLICE(PointerGetDatum(input), 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
 
 	double cr = (double) toast_raw_datum_size((Datum)input) / (double) (input_header->sequence_length + VARHDRSZ);
 
diff --git a/src/types/aligned_rna_sequence.c b/src/types/aligned_rna_sequence.c
index 76a657d..8a0e2d7 100644
--- a/src/types/aligned_rna_sequence.c
+++ b/src/types/aligned_rna_sequence.c
@@ -17,7 +17,7 @@
 #include "fmgr.h"
 #include "utils/array.h"
 #include "catalog/pg_type.h"
-#include "access/tuptoaster.h"
+#include "access/detoast.h"
 
 #include "sequence/sequence.h"
 #include "sequence/stats.h"
@@ -42,7 +42,7 @@ static PB_CodeSet aligned_rna_flc = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) true,
-		.fixed_id = (bool) 0,
+		.fixed_id = (uint8) 0,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 105553116266496,
 		.ascii_bitmap_high = (uint64) 2097290,
@@ -58,7 +58,7 @@ static PB_CodeSet aligned_rna_flc_cs = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) false,
-		.fixed_id = (bool) 1,
+		.fixed_id = (uint8) 1,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 105553116266496,
 		.ascii_bitmap_high = (uint64) 9007791962325130,
@@ -75,7 +75,7 @@ static PB_CodeSet aligned_rna_iupac = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) true,
-		.fixed_id = (bool) 2,
+		.fixed_id = (uint8) 2,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 105553116266496,
 		.ascii_bitmap_high = (uint64) 49047966,
@@ -95,7 +95,7 @@ static PB_CodeSet aligned_rna_iupac_cs = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) false,
-		.fixed_id = (bool) 3,
+		.fixed_id = (uint8) 3,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 105553116266496,
 		.ascii_bitmap_high = (uint64) 210659409954367902,
@@ -119,7 +119,7 @@ static PB_CodeSet aligned_rna_flc_complement = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) true,
-		.fixed_id = (bool) 4,
+		.fixed_id = (uint8) 4,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 105553116266496,
 		.ascii_bitmap_high = (uint64) 2097290,
@@ -135,7 +135,7 @@ static PB_CodeSet aligned_rna_flc_cs_complement = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) false,
-		.fixed_id = (bool) 5,
+		.fixed_id = (uint8) 5,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 105553116266496,
 		.ascii_bitmap_high = (uint64) 9007791962325130,
@@ -152,7 +152,7 @@ static PB_CodeSet aligned_rna_iupac_complement = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) true,
-		.fixed_id = (bool) 6,
+		.fixed_id = (uint8) 6,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 105553116266496,
 		.ascii_bitmap_high = (uint64) 49047966,
@@ -172,7 +172,7 @@ static PB_CodeSet aligned_rna_iupac_cs_complement = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) false,
-		.fixed_id = (bool) 7,
+		.fixed_id = (uint8) 7,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 105553116266496,
 		.ascii_bitmap_high = (uint64) 210659409954367902,
@@ -507,7 +507,7 @@ Datum aligned_rna_sequence_typmod_out(PG_FUNCTION_ARGS)
 
 	PB_TRACE(errmsg("<-aligned_rna_sequence_typmod_out() returning '%s'", out));
 
-	PG_RETURN_INT32(result);
+	PG_RETURN_CSTRING(result);
 }
 
 /**
@@ -727,7 +727,7 @@ Datum aligned_rna_sequence_substring (PG_FUNCTION_ARGS)
 	PB_TRACE(errmsg("->aligned_rna_sequence_substring()"));
 
 	input_header = (PB_CompressedSequence*)
-			PG_DETOAST_DATUM_SLICE(input, 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
+			PG_DETOAST_DATUM_SLICE(PointerGetDatum(input), 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
 
 	if (len < 0) {
 		ereport(ERROR,(errmsg("negative substring length not allowed")));
@@ -771,7 +771,7 @@ PG_FUNCTION_INFO_V1 (aligned_rna_sequence_char_length);
 Datum aligned_rna_sequence_char_length (PG_FUNCTION_ARGS)
 {
 	PB_CompressedSequence* seq = (PB_CompressedSequence*)
-			PG_DETOAST_DATUM_SLICE(PG_GETARG_RAW_VARLENA_P(0),0,4);
+			PG_DETOAST_DATUM_SLICE(PG_GETARG_DATUM(0),0,4);
 
 	PG_RETURN_INT32(seq->sequence_length);
 }
@@ -791,7 +791,7 @@ Datum aligned_rna_sequence_compression_ratio (PG_FUNCTION_ARGS)
 {
 	Varlena* input = (Varlena*) PG_GETARG_RAW_VARLENA_P(0);
 	PB_CompressedSequence* input_header = (PB_CompressedSequence*)
-		PG_DETOAST_DATUM_SLICE(input, 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
+		PG_DETOAST_DATUM_SLICE(PointerGetDatum(input), 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
 
 	double cr = (double) toast_raw_datum_size((Datum)input) / (double) (input_header->sequence_length + VARHDRSZ);
 
@@ -1137,5 +1137,3 @@ Datum octet_length_aligned_rna(PG_FUNCTION_ARGS)
 {
 	PG_RETURN_UINT32((uint32) toast_raw_datum_size((Datum) PG_GETARG_RAW_VARLENA_P(0)));
 }
-
-
diff --git a/src/types/alphabet.c b/src/types/alphabet.c
index 6a5c1cc..b56b124 100644
--- a/src/types/alphabet.c
+++ b/src/types/alphabet.c
@@ -373,7 +373,7 @@ text* alphabet_to_text(PB_Alphabet* input)
 
 	ereport(DEBUG1,(errmsg("Alphabet size as text:%d", size)));
 
-	result = palloc(size);
+	result = palloc(size + 1);
 	SET_VARSIZE(result,size);
 	output_pointer = (char*) VARDATA(result);
 
@@ -466,11 +466,11 @@ Datum alphabet_in_text (PG_FUNCTION_ARGS)
 PG_FUNCTION_INFO_V1 (alphabet_out);
 Datum alphabet_out (PG_FUNCTION_ARGS) {
 	PB_Alphabet* input =
-			(PB_Alphabet*) PG_DETOAST_DATUM(PG_GETARG_POINTER(0));
+			(PB_Alphabet*) PG_DETOAST_DATUM(PG_GETARG_DATUM(0));
 
 	ereport(DEBUG1,(errmsg("as:%d", PB_ALPHABET_SIZE(input))));
 
-	PG_RETURN_CSTRING(TextDatumGetCString(alphabet_to_text(input)));
+	PG_RETURN_CSTRING(text_to_cstring(alphabet_to_text(input)));
 }
 
 /**
@@ -482,7 +482,7 @@ Datum alphabet_out (PG_FUNCTION_ARGS) {
 PG_FUNCTION_INFO_V1 (alphabet_out_text);
 Datum alphabet_out_text (PG_FUNCTION_ARGS) {
 	PB_Alphabet* input =
-			(PB_Alphabet*) PG_DETOAST_DATUM(PG_GETARG_POINTER(0));
+			(PB_Alphabet*) PG_DETOAST_DATUM(PG_GETARG_DATUM(0));
 
 	ereport(DEBUG1,(errmsg("as:%d", PB_ALPHABET_SIZE(input))));
 
@@ -567,4 +567,3 @@ PB_Alphabet* get_alphabet_compressed_sequence (PB_CompressedSequence* input, PB_
 	return result;
 }
 
-
diff --git a/src/types/dna_sequence.c b/src/types/dna_sequence.c
index 0929589..455c9d5 100644
--- a/src/types/dna_sequence.c
+++ b/src/types/dna_sequence.c
@@ -17,7 +17,7 @@
 #include "fmgr.h"
 #include "utils/array.h"
 #include "catalog/pg_type.h"
-#include "access/tuptoaster.h"
+#include "access/detoast.h"
 
 #include "sequence/sequence.h"
 #include "sequence/stats.h"
@@ -41,7 +41,7 @@ static PB_CodeSet dna_flc = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) true,
-		.fixed_id = (bool) 0,
+		.fixed_id = (uint8) 0,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 0,
 		.ascii_bitmap_high = (uint64) 1048714,
@@ -57,7 +57,7 @@ static PB_CodeSet dna_flc_cs = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) false,
-		.fixed_id = (bool) 1,
+		.fixed_id = (uint8) 1,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 0,
 		.ascii_bitmap_high = (uint64) 4504192333906058,
@@ -74,7 +74,7 @@ static PB_CodeSet dna_iupac = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) true,
-		.fixed_id = (bool) 2,
+		.fixed_id = (uint8) 2,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 0,
 		.ascii_bitmap_high = (uint64) 47999390,
@@ -93,7 +93,7 @@ static PB_CodeSet dna_iupac_cs = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) false,
-		.fixed_id = (bool) 3,
+		.fixed_id = (uint8) 3,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 0,
 		.ascii_bitmap_high = (uint64) 206155810325948830,
@@ -116,7 +116,7 @@ static PB_CodeSet dna_flc_complement = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) true,
-		.fixed_id = (bool) 4,
+		.fixed_id = (uint8) 4,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 0,
 		.ascii_bitmap_high = (uint64) 1048714,
@@ -132,7 +132,7 @@ static PB_CodeSet dna_flc_cs_complement = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) false,
-		.fixed_id = (bool) 5,
+		.fixed_id = (uint8) 5,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 0,
 		.ascii_bitmap_high = (uint64) 4504192333906058,
@@ -149,7 +149,7 @@ static PB_CodeSet dna_iupac_complement = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) true,
-		.fixed_id = (bool) 6,
+		.fixed_id = (uint8) 6,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 0,
 		.ascii_bitmap_high = (uint64) 47999390,
@@ -168,7 +168,7 @@ static PB_CodeSet dna_iupac_cs_complement = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) false,
-		.fixed_id = (bool) 7,
+		.fixed_id = (uint8) 7,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 0,
 		.ascii_bitmap_high = (uint64) 206155810325948830,
@@ -809,7 +809,7 @@ Datum dna_sequence_substring (PG_FUNCTION_ARGS)
 	PB_TRACE(errmsg("->dna_sequence_substring()"));
 
 	input_header = (PB_CompressedSequence*)
-			PG_DETOAST_DATUM_SLICE(input, 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
+			PG_DETOAST_DATUM_SLICE(PointerGetDatum(input), 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
 
 	if (len < 0) {
 		ereport(ERROR,(errmsg("negative substring length not allowed")));
@@ -853,7 +853,7 @@ PG_FUNCTION_INFO_V1 (dna_sequence_char_length);
 Datum dna_sequence_char_length (PG_FUNCTION_ARGS)
 {
 	PB_CompressedSequence* dna_seq = (PB_CompressedSequence*)
-			PG_DETOAST_DATUM_SLICE(PG_GETARG_RAW_VARLENA_P(0),0,4);
+			PG_DETOAST_DATUM_SLICE(PG_GETARG_DATUM(0),0,4);
 
 	PG_RETURN_INT32(dna_seq->sequence_length);
 }
@@ -873,7 +873,7 @@ Datum dna_sequence_compression_ratio (PG_FUNCTION_ARGS)
 {
 	Varlena* input = (Varlena*) PG_GETARG_RAW_VARLENA_P(0);
 	PB_CompressedSequence* input_header = (PB_CompressedSequence*)
-		PG_DETOAST_DATUM_SLICE(input, 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
+		PG_DETOAST_DATUM_SLICE(PointerGetDatum(input), 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
 
 	double cr = (double) toast_raw_datum_size((Datum)input) / (double) (input_header->sequence_length + VARHDRSZ);
 
diff --git a/src/types/rna_sequence.c b/src/types/rna_sequence.c
index 32ec2aa..5757876 100644
--- a/src/types/rna_sequence.c
+++ b/src/types/rna_sequence.c
@@ -17,7 +17,7 @@
 #include "fmgr.h"
 #include "utils/array.h"
 #include "catalog/pg_type.h"
-#include "access/tuptoaster.h"
+#include "access/detoast.h"
 
 #include "sequence/sequence.h"
 #include "sequence/stats.h"
@@ -41,7 +41,7 @@ static PB_CodeSet rna_flc = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) true,
-		.fixed_id = (bool) 0,
+		.fixed_id = (uint8) 0,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 0,
 		.ascii_bitmap_high = (uint64) 2097290,
@@ -57,7 +57,7 @@ static PB_CodeSet rna_flc_cs = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) false,
-		.fixed_id = (bool) 1,
+		.fixed_id = (uint8) 1,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 0,
 		.ascii_bitmap_high = (uint64) 9007791962325130,
@@ -74,7 +74,7 @@ static PB_CodeSet rna_iupac = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) true,
-		.fixed_id = (bool) 2,
+		.fixed_id = (uint8) 2,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 0,
 		.ascii_bitmap_high = (uint64) 49047966,
@@ -93,7 +93,7 @@ static PB_CodeSet rna_iupac_cs = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) false,
-		.fixed_id = (bool) 3,
+		.fixed_id = (uint8) 3,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 0,
 		.ascii_bitmap_high = (uint64) 210659409954367902,
@@ -116,7 +116,7 @@ static PB_CodeSet rna_flc_complement = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) true,
-		.fixed_id = (bool) 4,
+		.fixed_id = (uint8) 4,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 0,
 		.ascii_bitmap_high = (uint64) 2097290,
@@ -132,7 +132,7 @@ static PB_CodeSet rna_flc_cs_complement = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) false,
-		.fixed_id = (bool) 5,
+		.fixed_id = (uint8) 5,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 0,
 		.ascii_bitmap_high = (uint64) 9007791962325130,
@@ -149,7 +149,7 @@ static PB_CodeSet rna_iupac_complement = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) true,
-		.fixed_id = (bool) 6,
+		.fixed_id = (uint8) 6,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 0,
 		.ascii_bitmap_high = (uint64) 49047966,
@@ -168,7 +168,7 @@ static PB_CodeSet rna_iupac_cs_complement = {
 		.is_fixed = (bool) true,
 		.uses_rle = (bool) false,
 		.ignore_case = (bool) false,
-		.fixed_id = (bool) 7,
+		.fixed_id = (uint8) 7,
 		.swap_savings = (uint64) 0,
 		.ascii_bitmap_low = (uint64) 0,
 		.ascii_bitmap_high = (uint64) 210659409954367902,
@@ -314,7 +314,8 @@ PB_CompressedSequence* compress_rna_sequence(uint8* input,
 		/*
 		 * Choose applicable pre-built code set.
 		 */
-		if (typmod.restricting_alphabet == PB_RNA_TYPMOD_IUPAC) {
+		if (typmod.restricting_alphabet == PB_RNA_TYPMOD_IUPAC ||
+			!PB_CHECK_CODESET((&rna_flc_cs), info)) {
 			if (typmod.case_sensitive == PB_RNA_TYPMOD_CASE_SENSITIVE) {
 				fixed_code_id = 3;
 			} else {
@@ -522,7 +523,7 @@ Datum rna_sequence_typmod_out(PG_FUNCTION_ARGS) {
 	out++;
 	*out = 0;
 
-	PG_RETURN_INT32(result);
+	PG_RETURN_CSTRING(result);
 }
 
 /**
@@ -740,7 +741,7 @@ Datum rna_sequence_substring (PG_FUNCTION_ARGS)
 	PB_TRACE(errmsg("->rna_sequence_substring()"));
 
 	input_header = (PB_CompressedSequence*)
-			PG_DETOAST_DATUM_SLICE(input, 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
+			PG_DETOAST_DATUM_SLICE(PointerGetDatum(input), 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
 
 	if (len < 0) {
 		ereport(ERROR,(errmsg("negative substring length not allowed")));
@@ -783,7 +784,7 @@ Datum rna_sequence_substring (PG_FUNCTION_ARGS)
 PG_FUNCTION_INFO_V1 (rna_sequence_char_length);
 Datum rna_sequence_char_length (PG_FUNCTION_ARGS) {
 	PB_CompressedSequence* rna_seq = (PB_CompressedSequence*)
-			PG_DETOAST_DATUM_SLICE(PG_GETARG_RAW_VARLENA_P(0),0,4);
+			PG_DETOAST_DATUM_SLICE(PG_GETARG_DATUM(0),0,4);
 
 	PG_RETURN_INT32(rna_seq->sequence_length);
 }
@@ -802,7 +803,7 @@ PG_FUNCTION_INFO_V1 (rna_sequence_compression_ratio);
 Datum rna_sequence_compression_ratio (PG_FUNCTION_ARGS) {
 	Varlena* input = (Varlena*) PG_GETARG_RAW_VARLENA_P(0);
 	PB_CompressedSequence* input_header = (PB_CompressedSequence*)
-		PG_DETOAST_DATUM_SLICE(input, 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
+		PG_DETOAST_DATUM_SLICE(PointerGetDatum(input), 0, sizeof(PB_CompressedSequence) - VARHDRSZ);
 
 	double cr = (double) toast_raw_datum_size((Datum)input) / (double) (input_header->sequence_length + VARHDRSZ);
 
@@ -1184,4 +1185,3 @@ Datum octet_length_rna(PG_FUNCTION_ARGS)
 {
 	PG_RETURN_UINT32((uint32) toast_raw_datum_size((Datum) PG_GETARG_RAW_VARLENA_P(0)));
 }
-
