--- a/rdf_fdw.c
+++ b/rdf_fdw.c
@@ -823,8 +823,13 @@ Datum rdf_fdw_settings(PG_FUNCTION_ARGS)
 		appendStringInfo(&buffer, "zlib %s,", ver->libz_version);
 	if (ver->libssh_version)
 		appendStringInfo(&buffer, "libSSH %s,", ver->libssh_version);
+#if LIBCURL_VERSION_NUM >= 0x074200
 	if (ver->nghttp2_version)
 		appendStringInfo(&buffer, "nghttp2 %s,", ver->nghttp2_version);
+#elif defined(CURL_VERSION_HTTP2)
+	if (ver->features & CURL_VERSION_HTTP2)
+		appendStringInfoString(&buffer, "HTTP2,");
+#endif
 
 #ifdef RDF_FDW_CC
 	appendStringInfo(&buffer, "compiled by %s,", RDF_FDW_CC);
