--- a/wrappers/Cargo.toml
+++ b/wrappers/Cargo.toml
@@ -234,6 +234,6 @@
 ]
 [dependencies]
-pgrx = { version = "=0.16.1" }
+pgrx = { version = "=0.18.1" }
 #supabase-wrappers = "0.1"
 supabase-wrappers = { path = "../supabase-wrappers", default-features = false }
 
@@ -358,2 +358,2 @@
 [dev-dependencies]
-pgrx-tests = "=0.16.1"
+pgrx-tests = "=0.18.1"
--- a/supabase-wrappers/Cargo.toml
+++ b/supabase-wrappers/Cargo.toml
@@ -25,14 +25,14 @@
 pg_test = []
 
 [dependencies]
-pgrx = { version = "=0.16.1", default-features = false }
+pgrx = { version = "=0.18.1", default-features = false }
 thiserror = "1.0.63"
 tokio = { version = "1.43", features = ["rt", "net"] }
 uuid = { version = "1.10.0" }
 supabase-wrappers-macros = { version = "0.1", path = "../supabase-wrappers-macros" }
 
 [dev-dependencies]
-pgrx-tests = "=0.16.1"
+pgrx-tests = "=0.18.1"
 
 [package.metadata.docs.rs]
 features = ["pg15"]
--- a/supabase-wrappers/src/utils.rs
+++ b/supabase-wrappers/src/utils.rs
@@ -242,7 +242,7 @@
     ereport!(
         PgLogLevel::INFO,
         PgSqlErrorCode::ERRCODE_SUCCESSFUL_COMPLETION,
-        msg,
+        msg.to_string(),
         "Wrappers"
     );
 }
@@ -262,7 +262,7 @@
     ereport!(
         PgLogLevel::NOTICE,
         PgSqlErrorCode::ERRCODE_SUCCESSFUL_COMPLETION,
-        msg,
+        msg.to_string(),
         "Wrappers"
     );
 }
@@ -282,7 +282,7 @@
     ereport!(
         PgLogLevel::WARNING,
         PgSqlErrorCode::ERRCODE_WARNING,
-        msg,
+        msg.to_string(),
         "Wrappers"
     );
 }
@@ -307,3 +307,3 @@
 pub fn report_error(code: PgSqlErrorCode, msg: &str) {
-    ereport!(PgLogLevel::ERROR, code, msg, "Wrappers");
+    ereport!(PgLogLevel::ERROR, code, msg.to_string(), "Wrappers");
 }
